### ISSUE TITLE
Non-root install?

### ISSUE THREAD

**Issue by basicfeatures** [solidusio/solidus_starter_frontend#271]
Hello! Trying to add the starter frontend to my namespaced Rails app but it seems the installer places everything into the root so I have to manually move things around afterwards. Maybe in the future it could support something like this?

`rails generate solidus:install --frontend=solidus_starter_frontend --namespace=store`

Thanks 🙏

**Comment by waiting-for-dev** [comment]
Thanks for your feedback, @basicfeatures. That's something present in our roadmap but not there yet 🙂

**Comment by elia** [comment]
@basicfeatures would you be able to provide the basic steps (need for it to be fancy!) you used to move manually from root to a namespace? 

That could turn very useful for: 
1. anyone wanting to take up the challenge and submit a PR addressing this feature 
2. other people having the same need, at least until a better solution is in place 🙏

**Comment by basicfeatures** [comment]
Hey guys,

Thanks for the interest!

Namespacing my app has been quite the hassle, but definitely worth it as things are so much cleaner and better organized now

My Solidus namespace is called `Marketplace`, and so far I've:
```
cd ~/mylovelyapp/app/views/
mkdir marketplace/
mv address/ checkouts/ cart_line_items/ layouts/ home/ products/ orders/ taxons/ share/ marketplace/

cd ~/mylovelyapp/app/controllers
mkdir marketplace
mv cart_line_items_controller.rb checkouts_controller.rb coupon_codes_controller.rb home_controller.rb locale_controller.rb orders_controller.rb products_controller.rb store_controller.rb store_devise_controller.rb taxons_controller.rb marketplace/
cd marketplace/
replace "class " "class Marketplace::"
replace "StoreController" "Marketplace::StoreController"

cd ~/mylovelyapp/app/helpers/
mkdir marketplace/
mv checkout_helper.rb orders_helper.rb product_options_helper.rb shopping_bag_helper.rb taxon_filters_helper.rb taxons_helper.rb marketplace/
cd marketplace
replace "module " "module Marketplace::"
```

[`replace.sh`](https://gist.github.com/basicfeatures/09ec2f32a809b26ac520a09a2f493075) is a neat little shell script I wrote a while back.

Since I'm having a shared user database, I kept the `solidus_auth_devise` files in place, and instead told it to use my existing `User` model:

    rails generate spree:custom_user User && rails db:migrate

Then I wrapped the relevant routes inside my new namespace:
```
# config/routes.rb
mount Spree::Core::Engine, at: "/marketplace"
get "/marketplace", to: "marketplace/home#index"

namespace :marketplace do
  # mount Spree::Core::Engine, at: "/"
  # root "home#index"

  resource :account, controller: "users"
  resources :products, only: [:index, :show]
  resources :cart_line_items, only: :create

  [...]
```

I suspect we'll be needing to decorate some model associations too.

I used Andrew Culver's [My Method for Rails Model Namespacing](https://blog.bullettrain.co/rails-model-namespacing/) as my initial guide.

**Comment by fthobe** [comment]
Hey @basicfeatures are you still interested in this ticket?

**Comment by jarednorman** [comment]
No, just because everything is added to the host app does not mean that there's a way to install somewhere other than root. You would have to do the work yourself _and_ I don't know how much work that is. If it's insignificant, then we should document the process and close this. If it's significant, someone should eventually add support.

**Comment by fthobe** [comment]
Hey @basicfeatures , you opened up this ticket some time ago, did manage to workout some solution?

**Comment by jarednorman** [comment]
Sure. I'm sure someone will want this eventually, but realistically no one will do it until someone _needs_ it.
